include/zephyr/fs/fs.h

```diff
@@ -262,8 +262,11 @@ static inline void fs_dir_t_init(struct fs_dir_t *zdp)
  *   - @c FS_O_CREATE create file if it does not exist
  *   - @c FS_O_APPEND move to end of file before each write
  *
- * If @p flags are set to 0 the function will attempt to open an existing file
- * with no read/write access; this may be used to e.g. check if the file exists.
+ * If @p flags are set to 0 the function will attempt to open file with
+ * no read/write access; although this could be used to check if a file exists,
+ * this will, when logging is enabled for file systems, log error when file
+ * would not be found and will require file to be closed in case when found.
+ * The fs_stat should be used for checking whether file exists or not.
```

Comment:
If the only anticipated usage of using no flag is a makeshift and error-prone equivalent of fs_stat then maybe the warning should be made more to the point by saying something like the below? (also making it a more visible `@warning`

```suggestion
 * @warning If @p flags are set to 0 the function will attempt to open file with
            no read/write access. While this can be used to check if a file
            exists, it is recommended to use @see fs_stat instead.
```